src/main/java/org/checkerframework/checker/dependencyinjection/DependencyInjectionTransfer.java

```diff
@@ -41,8 +48,59 @@ public TransferResult<CFValue, CFStore> visitMethodInvocation(
               boundClassTypeMirror.getAnnotation(), diATF.classValValueElement, String.class);
 
       accumulate(node, result, classNames.toArray(new String[1]));
+
+    } else if (diATF.isAnnotatedWithMethod(node.getTree())) {
+
+      MethodInvocationNode methodInvocationNode = node;
+
+      CFValue value =
+          diATF
+              .getStoreBefore(node)
+              .getValue(JavaExpression.fromNode(methodInvocationNode.getTarget().getReceiver()));
+
+      if (value != null && !value.getAnnotations().isEmpty()) {
+        AnnotationMirrorSet annotations = value.getAnnotations();
+        annotations.forEach(
+            (annotation) -> {
+              if (AnnotationUtils.areSameByName(annotation, Bind.NAME)) {
+                List<String> classNames =
+                    AnnotationUtils.getElementValueArray(
+                        annotation, diATF.bindValValueElement, String.class);
+                MethodInvocationNode namesClassNode =
+                    (MethodInvocationNode) methodInvocationNode.getArgument(0);
+                StringLiteralNode givenName = (StringLiteralNode) namesClassNode.getArgument(0);
+
+                accumulateBindAnnotatedWith(node, result, classNames.get(0), givenName.getValue());
+              }
+            });
+      }
     }
 
     return result;
   }
+
+  public void accumulateBindAnnotatedWith(
+      Node node, TransferResult<CFValue, CFStore> result, String value, String name) {
+
+    JavaExpression target = JavaExpression.fromNode(node);
+
+    AnnotationMirror newAnno = diATF.createBindAnnotatedWithAnnotation(value, name);
+
+    // TODO: insertValue not updating stores, insertValuePermitNondeterministic is
+    if (result.containsTwoStores()) {
+      result.getThenStore().insertValuePermitNondeterministic(target, newAnno);
+      result.getElseStore().insertValuePermitNondeterministic(target, newAnno);
+    } else {
+      result.getRegularStore().insertValuePermitNondeterministic(target, newAnno);
```

Comment:
These calls still need to change.

Reply:
Calls to `insertIntoStores` is working as intended now! I believe it has to do with the most recent additions we made to our astub (`Names`).